### ISSUE TITLE
Too many false positives

### ISSUE THREAD

**Issue by ArneBachmannDLR** [tonybaloney/perflint#17]
Great project! I found at least one line of code that could be improved (moved loop to a dictionary expression); but I see, however, 99.5% false positives, e.g.:
- `loop-try-except-usage` can only be moved outside the loop, if the except clause contains a `break`. Otherwise I want to keep processing the loop
- `loop-invariant-statement` is usually not at all invariant but depends on a loop variable
- `use-tuple-over-list` is most often typed as a `List[...]` and therefore cannot and should not hold a tuple, even if initiated empty

**Comment by adamzev** [comment]
I enjoyed the talk and checking out the tool!

I ran into some issues with Pandas treating lists differently than tuples (when using them as keys).

I also hit a lot of false positives with `loop-invariant-statements` (unless I'm missing something).

I tried to simplify out an example of something it'd hit a false positive on:
```
keys = []
for record in ({'a': 4}, {'b': 8}):
    keys.append(tuple(k for k in record if k in ('b', 'c')))

print(keys)
```
I think it was saying ` k in ('b', 'c')` was loop invariant.
